src/Extensions/ArchiveRestoreAction.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace SilverStripe\VersionedAdmin\Extensions;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Versioned\RestoreAction;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\VersionedAdmin\ArchiveAdmin;
+
+/**
+ * Adds a restore action to the item edit form of ArchiveAdmin
+ */
+class ArchiveRestoreAction extends DataExtension
+{
+    /**
+     * Updates the edit form with a restore button if it is being viewed
+     *
+     * @param Form $form
+     * @return mixed
+     * @throws LogicException If the record is not versioned
+     */
+    public function updateItemEditForm(Form $form)
+    {
+        $record = $this->owner->getRecord();
+
+        if (!DataObject::has_extension($record, Versioned::class)) {
+            throw new LogicException(
```

Comment:
Sorry, there was two comments on this one. When you installed `Versioned`, `SilverStripe\Versioned\VersionedGridFieldItemRequest` automatically gets added to all gridfields, even if they are tracking listing non-versioned DO.

If we throw an exception here, it will break gridfields that track regular DO. Try to view a Member under Security to see what I'm talking about.